components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java

```diff
@@ -1469,4 +1469,11 @@ private boolean validateUpdateOfIdPEntityId(FederatedAuthenticatorConfig[] curre
         return true;
     }
 
+    public FederatedAuthenticatorConfig getFederatedAuthenticatorConfig(String idpName, String tenantDomain)
+            throws IdentityApplicationManagementException {
+        FederatedAuthenticatorConfig federatedAuthenticatorConfig = dao.getFederatedAuthenticatorConfig(null,
+                idpName, getTenantIdOfDomain(tenantDomain), tenantDomain);
+        return federatedAuthenticatorConfig;
+    }
+
```

Comment:
No need to use separate method here to get default authenticator. The client can use getIdPByName and then pick the default from the returned object. Also the method name is not right, it doesn't contain "default" anywhere.